internal/service/sso/data_source_application.go

```diff
@@ -0,0 +1,1166 @@
+					),
+					stringvalidator.LengthAtLeast(attrMinLength),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "A string that specifies the description of the application.",
+				Computed:    true,
+			},
+			"enabled": schema.BoolAttribute{
+				Description: "A boolean that specifies whether the application is enabled in the environment.",
+				Computed:    true,
+			},
+			"tags": schema.SetAttribute{
+				Description: "An array that specifies the list of labels associated with the application.",
+				ElementType: types.StringType,
+				Computed:    true,
+			},
+			"login_page_url": schema.StringAttribute{
+				Description: "A string that specifies the custom login page URL for the application.",
+				Computed:    true,
+			},
+			"icon": schema.ListNestedAttribute{
+				Description: "The HREF and the ID for the application icon.",
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "The ID for the application icon.",
+							Computed:    true,
+						},
+						"href": schema.StringAttribute{
+							Description: "The HREF for the application icon.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"access_control_role_type": schema.StringAttribute{
+				Description: "A string that specifies the user role required to access the application. A user is an admin user if the user has one or more of the following roles: Organization Admin, Environment Admin, Identity Data Admin, or Client Application Developer.",
+				Computed:    true,
+			},
+			"access_control_group_options": schema.ListNestedAttribute{
+				Description: "Group access control settings.",
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"type": schema.StringAttribute{
+							Description: "A string that specifies the group type required to access the application.",
+							Computed:    true,
+						},
+						"groups": schema.SetAttribute{
+							Description: "A set that specifies the group IDs for the groups the actor must belong to for access to the application.",
+							ElementType: types.StringType,
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"hidden_from_app_portal": schema.BoolAttribute{
+				Description: "A boolean to specify whether the application is hidden in the application portal despite the configured group access policy.",
+				Computed:    true,
+			},
+			"external_link_options": schema.ListNestedAttribute{
+				Description: "External link application specific settings.",
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"home_page_url": schema.StringAttribute{
+							Description: "A string that specifies the custom home page URL for the application.  Both `http://` and `https://` URLs are permitted.",
```

Comment:
The `MarkdownDescription` property should be used for markdown fields, but this is why we enclose in the `framework.SchemaAttributeDescriptionFromMarkdown` function.  Example: https://github.com/pingidentity/terraform-provider-pingone/blob/677191a70df4229ca33f1db91d2503e09746800b/internal/service/risk/resource_risk_predictor.go#L573-L574